src/basics.md

```diff
@@ -277,6 +279,79 @@ fn main() {
 }
 ```
 
+[ex-rand-passwd]: #ex-rand-passwd
+<a name="ex-rand-passwd"></a>
+## Create random passwords from a set of alphanumeric characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters in the range `A-Z, a-z, 0-9`, with [`gen_ascii_chars`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+fn main() {
+    let s: String = thread_rng().gen_ascii_chars().take(30).collect();
+    println!("{}", s);
+}
+```
+
+[ex-rand-choose]: #ex-rand-choose
+<a name="ex-rand-choose"></a>
+## Create random passwords from a set of user-defined characters
+
+[![rand-badge]][rand] [![cat-os-badge]][cat-os]
+
+Randomly generates a string of given length ASCII characters with custom user-defined bytestring, with [`choose`].
+
+```rust
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+
+struct CustomGenerator<'a, R:'a> {
+    rng: &'a mut R,
```

Comment:
I'd probably go with owing the `rng` to make the lifetimes simpler. But I'll leave it up to you :)
We might go with having a refference to some outsude charset